### ISSUE TITLE
bun test React Testing Library multiple elements error

### ISSUE THREAD

**Issue by ryandi-tiket** [oven-sh/bun#10366]
### What version of Bun is running?

1.1.4+fbe2fe0c3

### What platform is your computer?

Darwin 23.4.0 arm64 arm

### What steps can reproduce the bug?

```sh
git clone https://github.com/ryandi-tiket/testing-bun-test

bun install

bun test
```

### What is the expected behavior?

Tests should pass.

### What do you see instead?

Second test fails, because there are multiple buttons rendered (the button from the first test lingers to the second test).

**Comment by ryandi-tiket** [comment]
My bad, `cleanup` is needed in between tests. I mistakenly thought this was a bun issue, because the same tests were running fine in Jest.

**Comment by hussain-nz** [comment]
Running tests with `jest` or `craco` commands seemed to do this automatically. I will put the error done to make it easier to find the solution here :).

My error: 
TestingLibraryElementError: Found multiple elements with the role .... etc

As Ryandi explained, RTL isn't cleaning up between tests.
I couldn't set the cleanup method globally (getting some errors) so I set it up in each test:
```
import {  cleanup } from '@testing-library/react';
import { beforeEach, describe, expect, it, jest } from 'bun:test';

describe(() => {
  beforeEach(() => cleanup());
  ...
});
```
